ledger/src/blockstore.rs

```diff
@@ -3737,7 +3737,10 @@ fn chain_new_slot_to_prev_slot(
     current_slot: Slot,
     current_slot_meta: &mut SlotMeta,
 ) {
-    prev_slot_meta.next_slots.push(current_slot);
```

Comment:
To my knowledge, this code is not new but the issue we are observing is new.
So the duplicates here might only be the symptoms, not the culprit.
Any idea why we are observing problematic duplicates here now? but not before?

Reply:
I wasn't able to find reasons, but figured this could be an ok bandaid for now. I can add a metric for when duplicates are attempted to be added for debugging though

Reply:
Sounds good, how about a `debug_assert` and a `datapoint_warn`?

Reply:
yeah, that sounds good